src/convert.jl

```diff
@@ -986,3 +986,37 @@ function convert(::Type{Zonotope},
     Z2 = translate(Z1, vector(am), share=true)
     return Z2
 end
+
+"""
+    convert(::Type{<:HParallelotope}, Z::AbstractZonotope)
+
+Converts a zonotopic set of order 1 into a parallelotope in constraint
+representation.
+
+### Input
+
+- `HParallelotope` -- type used for dispatch
+- `Z`        -- zonotopic set
+
+### Output
+
+A parallelotope in constraint representation.
+"""
+function convert(::Type{<:HParallelotope}, Z::AbstractZonotope)
+    @assert order(Z) == 1 "cannot convert a zonotope that is not of order 1 to"*
+                          " a parallelotope"
+    n = dim(Z)
+
+    constraints = constraints_list(Z)
+
+    D = Matrix{N}(undef, n, n)
+    c = Vector{N}(undef, 2n)
+    j = 1
+    @inbounds for i in 1:n
+        D[i, :] = constraints[j].a
+        c[i] = constraints[j].b
+        c[i+n] = constraints[j+1].b
+        j += 2
+    end
```

Comment:
Does this assume a structure of `constraints`? The input can be any user-defined zonotope, so you cannot rely on that.

Reply:
It assumes the structure given by `constraints_list` in https://github.com/JuliaReach/LazySets.jl/blob/3b737cff9c75701d58e16750e9d5626a0046e8e5/src/Interfaces/AbstractZonotope.jl#L511

Reply:
in that case im fine by restricting the signature `Zonotope`.

Reply:
Yes, we should restrict to `Zonotope` because for instance `AbstractHyperrectangle`s use a different implementation. Maybe even add a test that the implementation of `constraints_list(::Zonotope)` uses that order (all tests I saw in `unit_zonotope.jl` do not explicitly check the order).